driver-core/src/main/java/com/datastax/driver/core/Metadata.java

```diff
@@ -321,8 +323,25 @@ public static boolean isReservedCqlKeyword(String id) {
         if (current == null) {
             return Collections.emptySet();
         } else {
-            Set<Host> hosts = current.getReplicas(keyspace, range.getEnd());
-            return hosts == null ? Collections.<Host>emptySet() : hosts;
+            Set<Host> replicas = Sets.newHashSet();
+            Set<Host> allHosts = getAllHosts();
+            // for each host's range, if it intersects with the input range, include that range's replicas.
+            // this is a bit repetitive as we may evaluate the same ranges multiple times.
```

Comment:
As documented, this is not ideal since we could evaluate the same range multiple times, but couldn't find a direct way to get at a host's primary ranges.